system/services/email/EmailStyleInliner.cfc

```diff
@@ -45,8 +45,24 @@ component {
 			return fromCache;
 		}
 
+		arguments.html = trim( arguments.html );
+
 		var innerHtmlOnly = !FindNoCase( "</html>", arguments.html );
-		var doc           = _jsoup.parse( arguments.html );
+
+		// special cases of widget which only consist of a table cell or row without a wrapping table tag
+		// jsoup will remove the TD / TR tags in those cases, therefore adding now and stripping after processing
+		var isHtmlTableCell = innerHtmlOnly && ReFindNoCase( "^<td[^>]*>", arguments.html );
+		var isHtmlTableRow  = innerHtmlOnly && !isHtmlTableCell && ReFindNoCase( "^<tr[^>]*>", arguments.html );
+
+		// add dummy wrapping html table and row tags to make sure jsoup parsing works as expected
+		if ( isHtmlTableCell ) {
```

Comment:
I suggest adding an id to the first containing element in both of these, i.e.

```cfc
arguments.html = "<table><tbody id='_css-inliner-wrapper'> & ...
```

Then, if using this, you can get the inner html of this element from the `doc`, rather than doing regex at the end.